### ISSUE TITLE
Future plans for inclusion of scikit_learn.KerasClassifier, PyTorch, Pyro, and C++ implantation version?

### ISSUE THREAD

**Issue by chrisjcc** [shap/shap#265]
Hi Scott,

    Quite an awesome job by all those involved this project in providing such an interesting tool for model interpretation. I have a few question in mind which I have outline below, and look forward hearing back from anyone much more informed that I on this topic. 

- Are there any plans for the SHAP explainer to run on tf.keras.wrappers.scikit_learn.KerasClassifier objects? I notice when I tried to apply the SHAP explainer on a KerasClassifier object, I recieved a message indicated that it is not currently supported. Such support would make it easier to work with a given platform, i.e. scikit-learn, without needing to switch between them.

- I was wondering if it is possible run SHAP on PyTorch, or even Pyro models, if these libraries would be compatible? I'm new to these topic, but am quite interested in doing a study with SHAP in the context of probabilistic programming, in particular Bayesian deep neural networks. Mainly study it's general application.

- On a different note, would there be an easy or straightforward way to port the python SHAP library into C++ version. I at times work with a C++ built ML library, and was curious if there's a possibility to make it compatible with the other library. 

Thanks for taking the time to read this comment.

**Comment by slundberg** [comment]
Hey!

- That's a great idea. That seems like a good thing for the gradient and deep explainers to have. No plans right now, but it should be a simple change that just pulls out the layers just like a regular keras model object.

- PyTorch support has been kindly added to master for gradient explainer and deep explainer by @GabrielTseng. I am on travel right now so I haven't had a chance to finish verifying and test it for merging. I should note that both those approaches, while intended to be an improvement/extension over other available approaches, still are far from perfect and so need to be used with care.

- The core Tree SHAP implementation is already in C++. The rest of it is in python. I don't plan to port anything more than I have to for C++, and things like plotting don't make sense as a C++ library. If there is a clear need though I am open to PRs or comments.
